tests/phpunit/tests/functions/getNonCachedIds.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Test `_get_non_cached_ids()`.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Test class for `_get_non_cached_ids()`.
+ *
+ * @group cache
+ *
+ * @covers ::_get_non_cached_ids
+ */
+class Tests_Functions_GetNonCachedIds extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 57593
+	 */
+	public function test_uncached_valid_ids_should_be_unique() {
+		$object_id = 1;
+
+		$this->assertSame( array( $object_id ), _get_non_cached_ids( array( $object_id, $object_id, (string) $object_id ), 'fake-group' ), 'Duplicate object IDs should be removed.' );
+	}
+
+	/**
+	 * @ticket 57593
+	 *
+	 * @dataProvider data_valid_ids_should_be_returned_as_integers
+	 *
+	 * @param mixed $object_id The object id.
+	 */
+	public function test_valid_ids_should_be_returned_as_integers( $object_id ) {
+		$this->assertSame( array( (int) $object_id ), _get_non_cached_ids( array( $object_id ), 'fake-group' ), 'Object IDs should be returned as integers.' );
```

Comment:
```suggestion
		$this->assertSame(
			array( (int) $object_id ),
			_get_non_cached_ids( array( $object_id ), 'fake-group' ),
			'Object IDs should be returned as integers.'
		);
```
Minor formatting edit, I find lines that fit on the screen easier to read 🙂